Announce.php

```diff
@@ -103,4 +105,82 @@ function schema() {
 			array( 'UpdateFunction', 'delete_orphans_dismissals' ),
 		);
 	}
+
+	/**
+	 * Add the RESTful routes handled by this plugin.
+	 *
+	 * @param string $p_event_name The event name
+	 * @param array  $p_event_args The event arguments
+	 * @return void
+	 */
+	public function routes( $p_event_name, $p_event_args ) {
+		$t_app = $p_event_args['app'];
+		$t_plugin = $this;
+		$t_app->group(
+			plugin_route_group(),
+			function() use ( $t_app, $t_plugin ) {
+				$t_app->get( '/dismiss/{context_id}', [$t_plugin, 'route_dismiss'] );
+			}
+		);
+	}
+
+	/**
+	 * RESTful route for Announcement dismissal.
+	 *
+	 * Returned JSON structure
+	 *   - {string} title
+	 *   - {string} text
+	 *
+	 * @param Psr\Http\Message\ServerRequestInterface $request
+	 * @param Psr\Http\Message\ResponseInterface $response
+	 * @param array $args
+	 * @return Psr\Http\Message\ResponseInterface
+	 */
+	public function route_dismiss( $request, $response, $args) {
+		$t_timestamp = time();
+		$t_user_id = auth_get_current_user_id();
+
+		# Set the reference Context Id for dismissal
+		$t_context_id = (int)$args['context_id'];
+
+		# Make sure the message context actually exists
+		$t_context = AnnounceContext::load_by_id( $t_context_id );
+		if( !$t_context ) {
+			return $response->withStatus( HTTP_STATUS_BAD_REQUEST );
+		}
+
+		plugin_push_current( $this->basename );
```

Comment:
I suggest having push as the first statement and pop as the last before return, similar to what you had at `xmlhttprequest_plugin_announce_dismiss()`.  This avoids confusion in the future.  Hopefully we will remove the need for these at one point.

Reply:
Since pushing is only needed when calling plugin API functions and none are used before that point, the idea was to avoid having to call *plugin_pop_current()* before the return statement. I'll gladly revisit this if and when it is no longer necessary to push/pop.